src/Service/Proxy.php

```diff
@@ -92,8 +92,9 @@ public function download(string $package, string $version): void
 
         foreach ($this->decodeMetadata($package) as $packageData) {
             $lastDist = $packageData['dist'] ?? $lastDist;
-            if ($version === $packageData['version']) {
-                $this->filesystem->write($this->distPath($package, $lastDist['reference'], $lastDist['type']), $this->downloader->getContents($lastDist['url'])
+            $path = $this->distPath($package, $lastDist['reference'], $lastDist['type']);
+            if ($version === $packageData['version'] && !$this->filesystem->has($path)) {
```

Comment:
added check if version already exist (without this `$this->filesystem->write` will throw exception)